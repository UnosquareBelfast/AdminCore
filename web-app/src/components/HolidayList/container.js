import React, { Component } from 'react';
import { PropTypes as PT } from 'prop-types';
import { updateHoliday } from '../../services/holidayService';
import holidayStatus from '../../utilities/holidayStatus';
import Swal from 'sweetalert2';

export default Wrapped =>
  class extends Component {

    static propTypes = {
      holidays: PT.array,
    }

    constructor(props) {
      super(props);
      this.state = { holidays: [] };
    }

    componentWillUpdate(nextProps) {
      if (nextProps.holidays != this.props.holidays) {
        this.setState({holidays: nextProps.holidays});
      }
    }

    approveHoliday = holiday => {
      if (holiday.holidayStatusId === holidayStatus.APPROVED) {
        Swal({title: 'Alert', text: 'Holiday already approved.'});
        return;
      }
      const approvedHoliday = { ...holiday };
      approvedHoliday.holidayStatusId = holidayStatus.APPROVED;
      const holidayIndex = this.state.holidays.indexOf(holiday);
      updateHoliday(approvedHoliday)
        .then(() => {
          const holidays = [...this.state.holidays];
          holidays[holidayIndex].holidayStatusId = holidayStatus.APPROVED;
          this.setState({ holidays });
        })
        .catch(error => {
          const holidays = [...this.state.holidays];
          holidays[holidayIndex].holidayStatusId = holiday.holidayStatusId;
          this.setState({ holidays });
          
          Swal({
            title: 'Could not approve holiday',
            text: error.message,
            type: 'error',
          });
        });
    };

    rejectHoliday = holiday => {
      if (holiday.holidayStatusId === holidayStatus.REJECTED) {
        Swal({title: 'Alert', text: 'Holiday already rejected.'});
        return;
      }
      const rejectedHoliday = { ...holiday };
      rejectedHoliday.holidayStatusId = holidayStatus.REJECTED;
      const holidayIndex = this.state.holidays.indexOf(holiday);
      updateHoliday(rejectedHoliday)
        .then(() => {
          const holidays = [...this.state.holidays];
          holidays[holidayIndex].holidayStatusId = holidayStatus.REJECTED;
          this.setState({ holidays });
        })
        .catch(error => {
          const holidays = [...this.state.holidays];
          holidays[holidayIndex].holidayStatusId = holiday.holidayStatusId;
          this.setState({ holidays });

          Swal({
            title: 'Could not reject holiday',
            text: error.message,
            type: 'error',
          });
        });
    };

    render() {
      return (
        <Wrapped
          {...this.props}
          {...this.state}
          approveHoliday={this.approveHoliday}
          rejectHoliday={this.rejectHoliday}
        />
      );
    }
  };
