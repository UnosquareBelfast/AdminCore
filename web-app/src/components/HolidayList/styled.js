import styled from 'styled-components';

export const HolidayTable = styled.table`
  width: 100%;

  tr:nth-child(even) {
    background-color: ${props => props.theme.colours.lightgrey};
  }

  tr td:last-of-type {
    text-align: right;
  }

  th {
    text-align: left;
    text-transform:capitalize;
  }
`;

export const StatusDot = styled.div`
  display: inline-block;
  position: relative;
  top: -2.5px;
  margin-right: 7px;
  height: 7px;
  width: 7px;
  border-radius: 7px;
  background: ${({ theme, status }) => theme.holidayStatus[status]};
`;

export const ButtonStyle = styled.div`
  display: inline-block;

  button {
    border: none;
    padding: 3px 10px;
    margin: 2px 5px 2px 0;
    background: ${props => props.color || props.theme.colours.unoBlue};
    color: white;
    border-radius: 3px;
    cursor: pointer;
  }

  button:hover {
    background: ${props => props.hoverColor || props.theme.colours.darkBlue};;
  }
`;