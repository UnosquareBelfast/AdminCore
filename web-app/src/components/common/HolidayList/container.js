import React, { Component } from 'react';
import { PropTypes as PT } from 'prop-types';
import { getAllHolidays, updateHoliday } from '../../../services/holidayService';
import holidayStatus from '../../../utilities/holidayStatus';
import Swal from 'sweetalert2';

export default Wrapped =>
  class extends Component {

    static propTypes = {
      holidays: PT.array,
    }

    constructor(props) {
      super(props);
      this.state = { holidays: props.holidays };
    }

    render() {
      return (
        <Wrapped
          {...this.state}
          {...this.props}
          approveHoliday={this.approveHoliday}
          rejectHoliday={this.rejectHoliday}
        />
      );
    }
  };
