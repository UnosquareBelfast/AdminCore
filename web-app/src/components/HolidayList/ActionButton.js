import React from 'react';
import { PropTypes as PT } from 'prop-types';
import { ButtonStyle } from './styled';


export const ActionButton = ({holiday, action, label, color, hoverColor}) => (
  <ButtonStyle color={color} hoverColor={hoverColor}>
    <button
      onClick={() => action(holiday)}
    >
      {label}
    </button>
  </ButtonStyle>
);

ActionButton.propTypes = {
  holiday: PT.object,
  action: PT.func,
  label: PT.string,
  color: PT.string,
  hoverColor: PT.string,
};

const generateAction = (holiday, action, label, color, hoverColor) => 
  <ActionButton
    key={`${holiday.holidayId}${label}`}
    holiday={holiday}
    action={action}
    label={label}
    color={color}
    hoverColor={hoverColor}
  />;

export default generateAction;