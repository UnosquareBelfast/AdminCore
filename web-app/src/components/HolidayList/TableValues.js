import React, { Fragment } from 'react';
import { StatusDot } from './styled';
import { statusText } from '../../utilities/holidayStatus';

const status = {
  id: 'status',
  Header: 'Status',
  sortable: false,
  filterable: false,
  accessor: holiday => (
    <Fragment>
      <StatusDot status={holiday.holidayStatusId} />
      {statusText[holiday.holidayStatusId]}
    </Fragment>
  ),
};

const employee = {
  id: 'employee',
  Header: 'Employee',
  accessor: ({ employee }) => `${employee.forename} ${employee.surname}`,
  sortMethod: (a, b) => {
    if (a === b) {
      return 0;
    }
    const aReverse = a
      .split('')
      .reverse()
      .join('');
    const bReverse = b
      .split('')
      .reverse()
      .join('');
    return aReverse > bReverse ? 1 : -1;
  },
};

const startDate = {
  id: 'startDate',
  Header: 'Start Date',
  accessor: holiday => holiday.start.format('D MMM YYYY'),
};

const endDate = {
  id: 'endDate',
  Header: 'End Date',
  accessor: holiday => holiday.end.format('D MMM YYYY'),
};

const requestedDate = {
  id: 'requestedDate',
  Header: 'Requested Date',
  accessor: holiday => holiday.requested.format('D MMM YYYY'),
};

export default {
  status,
  startDate,
  endDate,
  requestedDate,
  employee,
};
