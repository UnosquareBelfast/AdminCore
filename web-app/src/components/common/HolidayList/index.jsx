import React from 'react';
import { PropTypes as PT } from 'prop-types';
import container from './container';
import { HolidayTable, StatusDot } from './styled';
import { statusText } from '../../../utilities/holidayStatus';

const buildTableHeaders = (columns) => {
  return columns.reduce((acc, column) => {
    return acc.concat(<th key={column}>{column}</th>);
  }, []);
};

const buildTableValues = (columns, holiday) => {
  const { forename, surname } = holiday.employee;
  const { holidayId, start, end, requested, holidayStatusId } = holiday;

  const values = {
    status: <td key={`${holidayId}status`}><StatusDot status={holidayStatusId} />{statusText[holidayStatusId]}</td>,
    employee: <td key={`${holidayId}employee`}>{`${forename} ${surname}`}</td>,
    start: <td key={`${holidayId}start}`}>{start.format('Do MMM YYYY')}</td>,
    end: <td key={`${holidayId}end}`}>{end.format('Do MMM YYYY')}</td>,
    requested: <td key={`${holidayId}requested`}>{requested.format('Do MMM YYYY')}</td>,
  };

  return columns.reduce((acc, column) => {
    return acc.concat(values[column]);
  }, []);
};

const buildTableActions = (props, holiday) => {
  const {actions, approveHoliday, rejectHoliday } = props;
  const { holidayId } = holiday;

  const values = {
    approve: <button key={`${holidayId}approve`} onClick={() => approveHoliday(holiday)}>Approve</button>,
    reject: <button key={`${holidayId}reject`} onClick={() => rejectHoliday(holiday)}>Reject</button>,
  };

  return actions.reduce((acc, action) => {
    return acc.concat(values[action]);
  }, []);
};


export const HolidayList = (props) => {
  const {holidays, columns} = props;
  return (
    !holidays || holidays.length === 0
      ? <p>There are no holidays to show</p>
      : <HolidayTable>
        <tbody>
          <tr>
            {buildTableHeaders(columns)}
          </tr>
          {holidays.map(holiday => {
            const { holidayId } = holiday;
            return (
              <tr key={holidayId}>
                {buildTableValues(columns, holiday)}
                <td>
                  {buildTableActions(props, holiday)}
                </td>
              </tr>
            );
          })}
        </tbody>
      </HolidayTable>
  );
};

HolidayList.propTypes = {
  holidays: PT.array,
  columns: PT.array.isRequired,
  actions: PT.array.isRequired,
  approveHoliday: PT.func,
  rejectHoliday: PT.func,
};

export default container(HolidayList);
