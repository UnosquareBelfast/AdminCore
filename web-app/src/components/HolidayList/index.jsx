import React from 'react';
import { PropTypes as PT } from 'prop-types';
import container from './container';
import ReactTable from 'react-table';
import TableValues  from './TableValues';
import ActionButton from './ActionButton';
import { theme } from '../../styled';
import { ActionWrap } from './styled';
import moment from 'moment';
import { isEmpty } from 'lodash';

const buildActions = (holiday, actions) => {
  const actionComponents = [];

  for (const action in actions) {
    actionComponents.push(
      <ActionButton
        key={action}
        holiday={holiday}
        action={actions[action]}
        label={action}
        color={action === 'reject' ? theme.colours.red : null}
        hoverColor={action === 'reject' ? theme.colours.darkRed : null}
      />
    );
  }

  return <ActionWrap>{actionComponents}</ActionWrap>;
};


const buildColumns = (columns, actions) => {
  const formattedColumns =  columns.reduce((acc, column) => {
    return acc.concat(TableValues[column]);
  }, []);

  if (isEmpty(actions)) return formattedColumns;

  formattedColumns.push({
    id: 'actions',
    sortable: false,
    filterable: false,
    'accessor': holiday => buildActions(holiday, actions),
  });

  return formattedColumns;
};


const renderTable = (holidays, columns, actions) => {
  const formattedColumns = buildColumns(columns, actions);
  return (
    <ReactTable
      data={holidays}
      columns={formattedColumns}
      defaultPageSize={10}
      filterable
      defaultFilterMethod={(filter, row) =>
        String(row[filter.id]).includes(filter.value)}
      defaultSortMethod={(a, b) => {
        const aMoment = new moment(`${a} 00:00:00 GMT`);
        const bMoment = new moment(`${b} 00:00:00 GMT`);
        return aMoment.isBefore(bMoment) ? 1 : -1;
      }}
    />
  );
};


export const HolidayList = (props) => {
  const {holidays, columns, actions} = props;
  return (
    !holidays || holidays.length === 0
      ? <p>There are no holidays to show</p>
      : renderTable(holidays, columns, actions)
  );
};

HolidayList.propTypes = {
  holidays: PT.array,
  columns: PT.array.isRequired,
  actions: PT.object,
};

export default container(HolidayList);
