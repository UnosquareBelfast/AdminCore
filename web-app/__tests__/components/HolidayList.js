import React from 'react';
import { shallow } from 'enzyme';
import { HolidayList } from '../../src/components/HolidayList/index';
import moment from 'moment';

const mockHolidays = [
  {
    holidayId: 0,
    employee: {forename: 'Bob', surname: 'Newton'},
    start: new moment([2018, 1, 2]),
    end: new moment([2018, 3, 4]),
    requested: new moment([2018, 5, 24]),
    holidayStatusId: 1,
  },
  {
    holidayId: 1,
    employee: {forename: 'Jim', surname: 'Bob'},
    start: new moment([2018, 12, 12]),
    end: new moment([2018, 11, 11]),
    requested: new moment([2018, 5, 24]),
    holidayStatusId: 2,
  },
];

const mockColumns = ['status', 'employee', 'startDate', 'endDate', 'requestedDate'];
const mockActions = {'approve': jest.fn(), 'reject': jest.fn()};

describe('HolidayList', () => {
  let wrapper = shallow(
    <HolidayList
      holidays={mockHolidays}
      columns={mockColumns}
      actions={mockActions}
    />
  );

  it('Should match snapshot with actions and columns supplied', () => {
    expect(wrapper).toMatchSnapshot();
  });

  wrapper = shallow(
    <HolidayList
      holidays={mockHolidays}
      columns={mockColumns}
    />
  );

  it('Should match snapshot with no actions supplied', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
