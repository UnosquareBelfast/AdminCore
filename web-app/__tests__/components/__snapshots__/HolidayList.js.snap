// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HolidayList Should match snapshot with actions and columns supplied 1`] = `
<ReactTable
  AggregatedComponent={[Function]}
  ExpanderComponent={[Function]}
  FilterComponent={[Function]}
  LoadingComponent={[Function]}
  NoDataComponent={[Function]}
  PadRowComponent={[Function]}
  PaginationComponent={[Function]}
  PivotValueComponent={[Function]}
  ResizerComponent={[Function]}
  TableComponent={[Function]}
  TbodyComponent={[Function]}
  TdComponent={[Function]}
  TfootComponent={[Function]}
  ThComponent={[Function]}
  TheadComponent={[Function]}
  TrComponent={[Function]}
  TrGroupComponent={[Function]}
  aggregatedKey="_aggregated"
  className=""
  collapseOnDataChange={true}
  collapseOnPageChange={true}
  collapseOnSortingChange={true}
  column={
    Object {
      "Aggregated": undefined,
      "Cell": undefined,
      "Expander": undefined,
      "Filter": undefined,
      "Footer": undefined,
      "Header": undefined,
      "Pivot": undefined,
      "PivotValue": undefined,
      "aggregate": undefined,
      "className": "",
      "filterAll": false,
      "filterMethod": undefined,
      "filterable": undefined,
      "footerClassName": "",
      "footerStyle": Object {},
      "getFooterProps": [Function],
      "getHeaderProps": [Function],
      "getProps": [Function],
      "headerClassName": "",
      "headerStyle": Object {},
      "minWidth": 100,
      "resizable": undefined,
      "show": true,
      "sortMethod": undefined,
      "sortable": undefined,
      "style": Object {},
    }
  }
  columns={
    Array [
      Object {
        "Header": "Status",
        "accessor": [Function],
        "filterable": false,
        "id": "status",
        "sortable": false,
      },
      Object {
        "Header": "Employee",
        "accessor": [Function],
        "id": "employee",
        "sortMethod": [Function],
      },
      Object {
        "Header": "Start Date",
        "accessor": [Function],
        "id": "startDate",
      },
      Object {
        "Header": "End Date",
        "accessor": [Function],
        "id": "endDate",
      },
      Object {
        "Header": "Requested Date",
        "accessor": [Function],
        "id": "requestedDate",
      },
    ]
  }
  data={
    Array [
      Object {
        "employee": Object {
          "forename": "Bob",
          "surname": "Newton",
        },
        "end": "2018-04-03T23:00:00.000Z",
        "holidayId": 0,
        "holidayStatusId": 1,
        "requested": "2018-06-23T23:00:00.000Z",
        "start": "2018-02-02T00:00:00.000Z",
      },
      Object {
        "employee": Object {
          "forename": "Jim",
          "surname": "Bob",
        },
        "end": "2018-12-11T00:00:00.000Z",
        "holidayId": 1,
        "holidayStatusId": 2,
        "requested": "2018-06-23T23:00:00.000Z",
        "start": null,
      },
    ]
  }
  defaultExpanded={Object {}}
  defaultFilterMethod={[Function]}
  defaultFiltered={Array []}
  defaultPageSize={10}
  defaultResized={Array []}
  defaultSortDesc={false}
  defaultSortMethod={[Function]}
  defaultSorted={Array []}
  expanderDefaults={
    Object {
      "filterable": false,
      "resizable": false,
      "sortable": false,
      "width": 35,
    }
  }
  filterable={true}
  freezeWhenExpanded={false}
  getLoadingProps={[Function]}
  getNoDataProps={[Function]}
  getPaginationProps={[Function]}
  getProps={[Function]}
  getResizerProps={[Function]}
  getTableProps={[Function]}
  getTbodyProps={[Function]}
  getTdProps={[Function]}
  getTfootProps={[Function]}
  getTfootTdProps={[Function]}
  getTfootTrProps={[Function]}
  getTheadFilterProps={[Function]}
  getTheadFilterThProps={[Function]}
  getTheadFilterTrProps={[Function]}
  getTheadGroupProps={[Function]}
  getTheadGroupThProps={[Function]}
  getTheadGroupTrProps={[Function]}
  getTheadProps={[Function]}
  getTheadThProps={[Function]}
  getTheadTrProps={[Function]}
  getTrGroupProps={[Function]}
  getTrProps={[Function]}
  groupedByPivotKey="_groupedByPivot"
  indexKey="_index"
  loading={false}
  loadingText="Loading..."
  multiSort={true}
  nestingLevelKey="_nestingLevel"
  nextText="Next"
  noDataText="No rows found"
  ofText="of"
  onFetchData={[Function]}
  originalKey="_original"
  pageSizeOptions={
    Array [
      5,
      10,
      20,
      25,
      50,
      100,
    ]
  }
  pageText="Page"
  pivotDefaults={Object {}}
  pivotIDKey="_pivotID"
  pivotValKey="_pivotVal"
  previousText="Previous"
  resizable={true}
  resolveData={[Function]}
  rowsText="rows"
  showPageJump={true}
  showPageSizeOptions={true}
  showPagination={true}
  showPaginationBottom={true}
  showPaginationTop={false}
  sortable={true}
  style={Object {}}
  subRowsKey="_subRows"
/>
`;

exports[`HolidayList Should match snapshot with no actions supplied 1`] = `
<ReactTable
  AggregatedComponent={[Function]}
  ExpanderComponent={[Function]}
  FilterComponent={[Function]}
  LoadingComponent={[Function]}
  NoDataComponent={[Function]}
  PadRowComponent={[Function]}
  PaginationComponent={[Function]}
  PivotValueComponent={[Function]}
  ResizerComponent={[Function]}
  TableComponent={[Function]}
  TbodyComponent={[Function]}
  TdComponent={[Function]}
  TfootComponent={[Function]}
  ThComponent={[Function]}
  TheadComponent={[Function]}
  TrComponent={[Function]}
  TrGroupComponent={[Function]}
  aggregatedKey="_aggregated"
  className=""
  collapseOnDataChange={true}
  collapseOnPageChange={true}
  collapseOnSortingChange={true}
  column={
    Object {
      "Aggregated": undefined,
      "Cell": undefined,
      "Expander": undefined,
      "Filter": undefined,
      "Footer": undefined,
      "Header": undefined,
      "Pivot": undefined,
      "PivotValue": undefined,
      "aggregate": undefined,
      "className": "",
      "filterAll": false,
      "filterMethod": undefined,
      "filterable": undefined,
      "footerClassName": "",
      "footerStyle": Object {},
      "getFooterProps": [Function],
      "getHeaderProps": [Function],
      "getProps": [Function],
      "headerClassName": "",
      "headerStyle": Object {},
      "minWidth": 100,
      "resizable": undefined,
      "show": true,
      "sortMethod": undefined,
      "sortable": undefined,
      "style": Object {},
    }
  }
  columns={
    Array [
      Object {
        "Header": "Status",
        "accessor": [Function],
        "filterable": false,
        "id": "status",
        "sortable": false,
      },
      Object {
        "Header": "Employee",
        "accessor": [Function],
        "id": "employee",
        "sortMethod": [Function],
      },
      Object {
        "Header": "Start Date",
        "accessor": [Function],
        "id": "startDate",
      },
      Object {
        "Header": "End Date",
        "accessor": [Function],
        "id": "endDate",
      },
      Object {
        "Header": "Requested Date",
        "accessor": [Function],
        "id": "requestedDate",
      },
    ]
  }
  data={
    Array [
      Object {
        "employee": Object {
          "forename": "Bob",
          "surname": "Newton",
        },
        "end": "2018-04-03T23:00:00.000Z",
        "holidayId": 0,
        "holidayStatusId": 1,
        "requested": "2018-06-23T23:00:00.000Z",
        "start": "2018-02-02T00:00:00.000Z",
      },
      Object {
        "employee": Object {
          "forename": "Jim",
          "surname": "Bob",
        },
        "end": "2018-12-11T00:00:00.000Z",
        "holidayId": 1,
        "holidayStatusId": 2,
        "requested": "2018-06-23T23:00:00.000Z",
        "start": null,
      },
    ]
  }
  defaultExpanded={Object {}}
  defaultFilterMethod={[Function]}
  defaultFiltered={Array []}
  defaultPageSize={10}
  defaultResized={Array []}
  defaultSortDesc={false}
  defaultSortMethod={[Function]}
  defaultSorted={Array []}
  expanderDefaults={
    Object {
      "filterable": false,
      "resizable": false,
      "sortable": false,
      "width": 35,
    }
  }
  filterable={true}
  freezeWhenExpanded={false}
  getLoadingProps={[Function]}
  getNoDataProps={[Function]}
  getPaginationProps={[Function]}
  getProps={[Function]}
  getResizerProps={[Function]}
  getTableProps={[Function]}
  getTbodyProps={[Function]}
  getTdProps={[Function]}
  getTfootProps={[Function]}
  getTfootTdProps={[Function]}
  getTfootTrProps={[Function]}
  getTheadFilterProps={[Function]}
  getTheadFilterThProps={[Function]}
  getTheadFilterTrProps={[Function]}
  getTheadGroupProps={[Function]}
  getTheadGroupThProps={[Function]}
  getTheadGroupTrProps={[Function]}
  getTheadProps={[Function]}
  getTheadThProps={[Function]}
  getTheadTrProps={[Function]}
  getTrGroupProps={[Function]}
  getTrProps={[Function]}
  groupedByPivotKey="_groupedByPivot"
  indexKey="_index"
  loading={false}
  loadingText="Loading..."
  multiSort={true}
  nestingLevelKey="_nestingLevel"
  nextText="Next"
  noDataText="No rows found"
  ofText="of"
  onFetchData={[Function]}
  originalKey="_original"
  pageSizeOptions={
    Array [
      5,
      10,
      20,
      25,
      50,
      100,
    ]
  }
  pageText="Page"
  pivotDefaults={Object {}}
  pivotIDKey="_pivotID"
  pivotValKey="_pivotVal"
  previousText="Previous"
  resizable={true}
  resolveData={[Function]}
  rowsText="rows"
  showPageJump={true}
  showPageSizeOptions={true}
  showPagination={true}
  showPaginationBottom={true}
  showPaginationTop={false}
  sortable={true}
  style={Object {}}
  subRowsKey="_subRows"
/>
`;
